cmd/start.go

```diff
@@ -97,26 +105,44 @@ var startCmd = &cobra.Command{
 
 		rtLogger.Info(fmt.Sprintf("flagd version: %s (%s), built at: %s", Version, Commit, Date))
 
-		if viper.GetString(syncProviderFlagName) != "" {
-			rtLogger.Warn("DEPRECATED: The --sync-provider flag has been deprecated. " +
+		if viper.GetString(evaluatorFlagName) != "json" {
+			rtLogger.Warn("DEPRECATED: The --evaluator flag has been deprecated. " +
 				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
 		}
 
-		if viper.GetString(evaluatorFlagName) != "json" {
-			rtLogger.Warn("DEPRECATED: The --evaluator flag has been deprecated. " +
+		if viper.GetStringMapString(providerArgsFlagName) != nil {
+			rtLogger.Warn("DEPRECATED: The --sync-provider-args flag has been deprecated. " +
 				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
 		}
+
+		syncProviders, err := runtime.SyncProvidersFromURIs(viper.GetStringSlice(uriFlagName))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		syncProvidersFromConfig := []sync.ProviderConfig{}
+		if cfgFile == "" && viper.GetString(syncProvidersFlagName) != "" {
+			syncProvidersFromConfig, err = runtime.SyncProviderArgPass(viper.GetString(syncProvidersFlagName))
+			if err != nil {
+				log.Fatal(err)
```

Comment:
The feedback we get here seems fine. When I mucked with the JSON the errors were clear enough.